source/iommu/iommu.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iommu
+
+import (
+	"io/ioutil"
+
+	"github.com/golang/glog"
+)
+
+// Implement FeatureSource interface
+type Source struct{}
+
+func (s Source) Name() string { return "iommu" }
+
+func (s Source) Discover() ([]string, error) {
+	features := []string{}
+
+	// Check if any iommu devices are available
+	devices, err := ioutil.ReadDir("/sys/class/iommu/")
+	if err == nil {
+		if len(devices) > 0 {
+			glog.Errorf("Detected %v IOMMU devices", len(devices))
+			features = append(features, "enabled")
+		} else {
+			glog.Errorf("No IOMMU devices detected")
```

Comment:
This is not an error. Use Infof instead? May be it is not necessary to log this. It is sufficient if we log when we find a device.

Reply:
As with the previous case, Infof are not shown at all. It is debatable if this should be logged. There are two kinds of feature sources in nfd, some do not log anything if they fail to detect, some print something in any case. I kind of find it useful to see at least something in the log, to verify that the module was run successfully.